diff --git a/Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/Thread.cpp b/Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/Thread.cpp
--- a/Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/Thread.cpp
+++ b/Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/Thread.cpp
@@ -538,8 +538,8 @@
     case 0x1003: // ARCH_GET_FS
-      *reinterpret_cast<uint64_t*>(addr) = Frame->State.fs_cached;
+      *reinterpret_cast<uint64_t*>(FEX::HLE::G2H(addr)) = Frame->State.fs_cached;
       Result = 0;
       break;
     case 0x1004: // ARCH_GET_GS
-      *reinterpret_cast<uint64_t*>(addr) = Frame->State.gs_cached;
+      *reinterpret_cast<uint64_t*>(FEX::HLE::G2H(addr)) = Frame->State.gs_cached;
       Result = 0;
       break;
